### ISSUE TITLE
Subnet tags of AWS Cluster don't get removed when cluster is deleted

### ISSUE THREAD

**Issue by vishu2498** [kubernetes-sigs/cluster-api-provider-aws#3947]
/kind bug

**What steps did you take and what happened:**
1. Create a AWS cluster from cluster API which would use an already present VPC and subnet. (at this point, the subnets will now have the tags of the newly created cluster)
2. Delete the cluster.
3. After deletion of cluster, the subnet tags of the new cluster didn't get removed.
4. If user manually deletes that tag, then on restart of controller-manager deployment, the tags again get re-added.

**What did you expect to happen:**
1. Tags of new cluster should get deleted from subnet once cluster gets deleted.
2. These tags shouldn't get re-added on restart of controller-manager deployment.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: v2.0.2
- Kubernetes version: (use `kubectl version`): 1.22
- OS (e.g. from `/etc/os-release`):

**Comment by Ankitasw** [comment]
@vishu2498 if you could reproduce this again, could you please let me know the list of tags that gets applied to your subnets?

**Comment by wyike** [comment]
I am curious about  why "If user manually deletes that tag, then on restart of controller-manager deployment, the tags again get re-added" even after cluster has been removed ? Hi @Ankitasw  Do you know the reason?

**Comment by Ankitasw** [comment]
> I am curious about why "If user manually deletes that tag, then on restart of controller-manager deployment, the tags again get re-added" even after cluster has been removed ? Hi @Ankitasw Do you know the reason?

We dont have tag deletion in place for subnets while delete reconciliation, which is causing this issue. If you use BYO subnet, that subnet would get tagged again and again every time you use that subnet in your new cluster.
So thats why we have decided to go ahead with the proposal described on parent issue.

**Comment by wyike** [comment]
Indeed, even the clusters deleted, their tags are re-tagged when capa (restart) reconciles the shared subnet owned by its managed cluster.
Add a note for myself:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/network/subnets.go#L51 (get existing tags from awscluster spec instead of aws service)
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/network/subnets.go#L509 (re-tag all existing tags to the shared subnet when reconciling the managed cluster).
